### PR TITLE
[JENKINS-48743] - Add Jenkinsfile and enable FindBugs

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-48743

@reviewbybees @raul-arabaolaza

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -135,8 +135,26 @@
...
         </configuration>
       </plugin>
+      <!-- TODO: remove when library parent POM enables FindBugs by default -->
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <configuration>
+          <xmlOutput>true</xmlOutput>
+          <effort>Max</effort>
+          <!-- TODO: issues need to be fixed first-->
```
Comment (dwnusbaum) [inline pom.xml]: Is it worth adding `<threshold>Low</threshold>` (or whatever the correct config), or is that mostly just useful for catching serialization issues?

Comment (oleg-nenashev) [reply pom.xml]: I have started from that, but now it generates a long list of garbage issues like serialVersionID & Co. I think we need to fix them first.

File: pom.xml [hunk 2]
```diff
@@ -135,8 +135,26 @@
           <goals>deploy</goals>
         </configuration>
       </plugin>
+      <!-- TODO: remove when library parent POM enables FindBugs by default -->
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <configuration>
```
Comment (dwnusbaum) [inline pom.xml]: Should you add a version to the plugin?

    [WARNING] 
    [WARNING] Some problems were encountered while building the effective model for org.jenkins-ci.tests:plugins-compat-tester:jar:0.0.3-SNAPSHOT
    [WARNING] 'build.plugins.plugin.version' for org.codehaus.mojo:findbugs-maven-plugin is missing. @ org.jenkins-ci.tests:plugins-compat-tester-aggregator:0.0.3-SNAPSHOT, /Users/dnusbaum/Projects/plugin-compat-tester/pom.xml, line 139, column 15

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                }
+
+                stage('Archive') {
+                    /* Archive the test results */
+                    junit '**/target/surefire-reports/TEST-*.xml'
+
+                    if (label == 'linux') {
+                      archiveArtifacts artifacts: 'target/**/*.jar'
+                      // FindBugs is not enabled for this repo so far
+                      // findbugs pattern: '**/target/findbugsXml.xml'
```
Comment (dwnusbaum) [inline Jenkinsfile]: The Findbugs config has `<xmlOutput>true</xmlOutput>`, and running locally it does produce the reports. Can this be uncommented?

**Comment by raul-arabaolaza** [comment]
Not really sure about ignoring test failures but I guess can be changed later if PCT is stable enough

**Comment by oleg-nenashev** [comment]
INFRA ticket is here: https://issues.jenkins-ci.org/browse/INFRA-1452

**Comment by oleg-nenashev** [comment]
> Not really sure about ignoring test failures but I guess can be changed later if PCT is stable enough

Test failures will not be ignored. The build will be failed by JUnit in this case

**Comment by oleg-nenashev** [comment]
There is a mistake in Jenkinsfile:

```
17:39:32 [linux] ERROR: No artifacts found that match the file pattern "target/**/*.jar". Configuration error?
17:39:33 [linux] ERROR: ‘target/**/*.jar’ doesn’t match anything, but ‘**/*.jar’ does. Perhaps that’s what you mean?
```

It needs to be adjusted to support multi-module repos

**Pull request merged**